Models/Text/BytePairEncoding.swift

```diff
@@ -60,14 +59,66 @@ public struct BytePairEncoder {
 
         // Check if the new words parts are in the vocabulary, and backtrack if necessary.
         let encoded = parts.flatMap { part -> [String] in
-        if vocabulary.contains(part) { return [part] }
+            if vocabulary.contains(part) { return [part] }
             return splitRecursively(part)
         }
 
         // Update the cache and return.
         // if useCache { cache[token] = encoded }
         return encoded
     }
+
+    /// Encodes the provided token to a sequence of BPE-coded tokens.
+    ///
+    /// - Parameters:
+    ///   - token: Token to encode.
+    /// - Returns: Array containing the BPE-coded tokens.
+    public func encodeGpt2(token: String) -> [String] {
+        // Split into parts before encoding.
+        var parts = BytePairEncoder.splitWithDelimitersGpt2(
+            token: token,
+            glossaryRegex: BytePairEncoder.gpt2GlossaryRegex)
+        if parts.count < 2 {
+            // Encode the full token and return.
+            return parts.map({ BytePairEncoder.encodedToken($0) })
```

Comment:
```suggestion
            return parts.map { BytePairEncoder.encodedToken($0) }
```
Does the suggestion compile? If so, is that preferable?